piece.cc

```diff
@@ -39,11 +39,12 @@ int Piece::move(Piece *currentTile, Piece *targetTile, int newX, int newY) {
 
     */
 
-    updateLegalMoves();
+    std::map<Box, int> tempLegalMovesMap = updateLegalMoves();
+    legalMovesMap = &tempLegalMovesMap;
 
     const Box targetBox(newX, newY);
 
-    if (std::find(legalMovesArr->begin(),legalMovesArr->end(), targetBox) != legalMovesArr->end()) {
+    if (legalMovesMap->find(targetBox) != legalMovesMap->end()) {
```

Comment:
do u need to add the `std::` to the `find()` function here?

Reply:
No because this find isn't in the standard library it is a function for maps